app/Main.hs

```diff
@@ -8,19 +9,32 @@ import Data.Monoid ((<>))
 
 import Options.Applicative
 
-import Configuration.Dotenv (loadFile)
-
+import Control.Monad (void)
 import Control.Monad.IO.Class(MonadIO(..))
 
+import System.FilePath ((</>))
+import System.Directory (getCurrentDirectory)
 import System.Process (system)
 import System.Exit (exitWith)
 
+import Configuration.Dotenv
+import Configuration.Dotenv.Types
+
 data Options = Options
-  { files    :: [String]
-  , program  :: String
-  , overload :: Bool
+  { program           :: String -- ^ Program to run with the load env variables
+  , dotenvFile        :: String -- ^ Path for the .env file
+  , dotenvExampleFile :: String -- ^ Path for the .env.example file
+  , override          :: Bool   -- ^ Override current environment variables
   } deriving (Show)
 
+buildConfig :: FilePath -> FilePath -> Config
```

Comment:
Given that we do not export `Options`, I think this `buildConfig` function in redundant, we can just do everything in place in `dotEnv`. (I there is a different `buildConfig` in tests, that one is probably handy and can stay.)

One thing I'd like to see in SB Haskell code is lesser number of functions. When code is fragmented, it's more difficult to read. In this particular case, the `dotEnv` function is not necessary. When I read `main`, I just want to be able to read it all in one place. So why not write it this way:

```haskell
main :: IO ()
main = do
  Options {..} <- execParser opts -- opts could be a stand-alone top-level definition
  current <- getCurrentDirectory
  -- the rest goes here
```